core/src/main/java/brooklyn/management/internal/AbstractManagementContext.java

```diff
@@ -443,6 +441,22 @@ public void setManagementNodeUri(URI uri) {
         return uri;
     }
     
+    @Override
+    public CatalogInitialization getCatalogInitialization() {
+        if (catalogInitialization!=null) return catalogInitialization;
```

Comment:
Double-checked locking is broken in Java, unless you declare the variable volatile (http://www.cs.umd.edu/~pugh/java/memoryModel/DoubleCheckedLocking.html).

This code is not performance sensitive, so just delete this line to keep the code simpler.